atomic_reactor/utils/operator.py

```diff
@@ -25,6 +27,134 @@
 OPERATOR_CSV_KIND = "ClusterServiceVersion"
 
 
+def is_dict(obj):
+    """
+    Check if object is a dict or the ruamel.yaml equivalent of a dict
+    """
+    return isinstance(obj, (dict, CommentedMap))
+
+
+def is_list(obj):
+    """
+    Check if object is a list or the ruamel.yaml equivalent of a list
+    """
+    return isinstance(obj, (list, CommentedSeq))
+
+
+def is_str(obj):
+    """
+    Check if object is a string or bytes. On python 3, checking for string
+    would be sufficient, but on python 2, it may not be.
+    """
+    return isinstance(obj, (bytes, six.text_type))
+
+
+class PullspecRegex(object):
+    """
+    Regular expressions for all things related to pullspecs
+    """
+
+    # Alphanumeric characters
+    _alnum = r"[a-zA-Z0-9]"
+    # Characters that you might find in a registry, namespace, repo or tag
+    _name = r"[a-zA-Z0-9\-._]"
+    # Base16 characters
+    _base16 = r"[a-fA-F0-9]"
+
+    # A basic name is anything that contains only alphanumeric and name
+    # characters and starts and ends with an alphanumeric character
+    _basic_name = r"(?:(?:{alnum}{name}*{alnum})|{alnum})".format(alnum=_alnum, name=_name)
```

Comment:
in your regex, the 1st will cover 2+ chars  and 2nd 1 char
why not just ? (this will cover 1+ chars:
r"(?:{alnum}{name}*{alnum})"